TTTCore/AI.cs

```diff
@@ -2,46 +2,52 @@
 using System.Collections;
 using System.Collections.Generic;
 using System.Linq;
-using TTTCore;
 
-namespace MM.AI
+namespace ArtificialIntelligence
 {
     public class AI
     {
         private string OwnerToken { get; set; }
         private string OpponentToken { get; set; }
+        private IAlgorithm Algorithm { get; set; }
 
-        public AI(string ownerToken, string opponentToken)
+        public AI(
+            string ownerToken, string opponentToken, IBoardAnalyzer analyzer, IScorer scorer
+        )
         {
             this.OwnerToken = ownerToken;
             this.OpponentToken = opponentToken;
+            this.Algorithm = new Minimax(this.OwnerToken, this.OpponentToken);
```

Comment:
Perfect opportunity to inject the algorithm as opposed to instantiating it here.